src/baldr/incident_singleton.h

```diff
@@ -88,7 +53,10 @@ struct incident_singleton_t {
    */
   incident_singleton_t(const boost::property_tree::ptree& config,
                        const std::unordered_set<valhalla::baldr::GraphId>& tileset)
-      : state{new state_t{}}, watcher(watch, config, std::cref(tileset), state) {
+      : state{new state_t{}}, watcher(watch, config, tileset, state, interrupt()) {
+    // let the thread control its own lifetime
+    watcher.detach();
```

Comment:
we need to detach right away so that if we throw below the thread is already disowned